src/instana/instrumentation/urllib3.py

```diff
@@ -2,109 +2,137 @@
-
+                    span.set_attribute(
+                        f"http.header.{custom_header}", headers[custom_header]
+                    )
         except Exception:
-            logger.debug("extract_custom_headers: ", exc_info=True)
-
-
-    def collect(instance, args, kwargs):
-        """ Build and return a fully qualified URL for this request """
+            logger.debug("urllib3 _extract_custom_headers error: ", exc_info=True)
+
+    def _collect_kvs(
+        instance: Union[
+            urllib3.connectionpool.HTTPConnectionPool,
+            urllib3.connectionpool.HTTPSConnectionPool,
+        ],
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
         kvs = dict()
         try:
-            kvs['host'] = instance.host
-            kvs['port'] = instance.port
+            kvs["host"] = instance.host
+            kvs["port"] = instance.port
 
-            if args is not None and len(args) == 2:
-                kvs['method'] = args[0]
-                kvs['path'] = args[1]
+            if args and len(args) == 2:
+                kvs["method"] = args[0]
+                kvs["path"] = args[1]
             else:
-                kvs['method'] = kwargs.get('method')
-                kvs['path'] = kwargs.get('path')
-                if kvs['path'] is None:
-                    kvs['path'] = kwargs.get('url')
+                kvs["method"] = kwargs.get("method")
+                kvs["path"] = (
+                    kwargs.get("path") if kwargs.get("path") else kwargs.get("url")
+                )
 
             # Strip any secrets from potential query params
-            if kvs.get('path') is not None and ('?' in kvs['path']):
-                parts = kvs['path'].split('?')
-                kvs['path'] = parts[0]
+            if kvs.get("path") and ("?" in kvs["path"]):
+                parts = kvs["path"].split("?")
+                kvs["path"] = parts[0]
                 if len(parts) == 2:
-                    kvs['query'] = strip_secrets_from_query(parts[1], agent.options.secrets_matcher,
-                                                            agent.options.secrets_list)
-
-            if type(instance) is urllib3.connectionpool.HTTPSConnectionPool:
-                kvs['url'] = 'https://%s:%d%s' % (kvs['host'], kvs['port'], kvs['path'])
+                    kvs["query"] = strip_secrets_from_query(
+                        parts[1],
+                        agent.options.secrets_matcher,
+                        agent.options.secrets_list,
+                    )
+
+            if isinstance(instance, urllib3.connectionpool.HTTPSConnectionPool):
+                kvs["url"] = "https://%s:%d%s" % (kvs["host"], kvs["port"], kvs["path"])
             else:
-                kvs['url'] = 'http://%s:%d%s' % (kvs['host'], kvs['port'], kvs['path'])
+                kvs["url"] = "http://%s:%d%s" % (kvs["host"], kvs["port"], kvs["path"])
```

Comment:
For better readability, can we use f strings here?

Reply:
I had to work around this once the linter was not getting issues with the f-string, but the runtime was. It will be a bit different.